packages/adapter-cloudflare/src/worker.js

```diff
@@ -53,10 +53,18 @@ const worker = {
 			});
 		}
 
-		// Writes to Cache only if allowed & specified
-		pragma = res.headers.get('cache-control');
-		return pragma && res.ok ? Cache.save(req, res, context) : res;
+		// write to `Cache` only if response is not an error,
+		// let `Cache.save` handle the Cache-Control and Vary headers
+		return !is_error(res.status) ? Cache.save(req, res, context) : res;
```

Comment:
```suggestion
		return res.status >= 400 ? res : Cache.save(req, res, context);
```

Reply:
And remove the other lines. I can’t do multi line on phone

Reply:
Thanks! Done. Remove or revert the comment as well?